src/core_modules/capture-core/reducers/descriptions/form.reducerDescription.js

```diff
@@ -14,12 +14,21 @@ import getOrgUnitRootsKey from '../../components/D2Form/field/Components/OrgUnit
 import {
     set as setStoreRoots,
 } from '../../components/FormFields/New/Fields/OrgUnitField/orgUnitRoots.store';
+import { registrationFormActionTypes } from '../../components/Pages/New/RegistrationDataEntry/RegistrationDataEntry.actions';
 
 const removeFormData = (state, { payload: { formId } }) => {
     const remainingKeys = Object.keys(state).filter(key => !key.includes(formId));
     return remainingKeys.reduce((acc, key) => ({ ...acc, [key]: state[key] }), {});
 };
 
+// cleans up data entries that _start with_ dataEntryId
+const cleanUp = (state, { payload: { dataEntryId } }) => {
+    const newState = Object.keys(state).reduce((acc, curr) =>
+        (curr.startsWith(dataEntryId) ? { ...acc, [curr]: {} } : { ...acc }), {});
+
+    return newState;
```

Comment:
I don't think this one is ever triggered? But if it was triggered, I think it would wipe all properties of the state object.

Reply:
I thought so too, but when you do that the app crashes. In some cases we arent really check if the object is `undefined`. Therefore if I erase the object completely aka make it `undefined` the app will crash..

Reply:
I mean it should be possible to use without cleanup:

Object.keys(state).reduce((acc, curr) =>
        (curr.startsWith(dataEntryId) ? { ...acc, [curr]: {} } : { ...acc, **[curr]: state[curr]** }), {});

Reply:
Not sure I am following, what would then be the difference?

Reply:
Applied the changes you are suggesting, thanks